### PR TITLE
feat: strf-9245 Warn user if shopper default language file is missing or has missing keys

### PR DESCRIPTION
**Pull request by jairo-bc**
#### What?
As a Stencil CLI user,
When I run `stencil start` for a particular store,
I will receive a warning if the store's default shopper language does not have a language file within the theme.

#### Tickets / Documentation

-   [STRF-9245](https://jira.bigcommerce.com/browse/STRF-9245)

#### Screenshots (if appropriate)
Screenshot 2021-12-02 at 17 24 40

cc @bigcommerce/storefront-team

### CODE REVIEW

**Comment by jmwiese** [comment]
I don't quite understand the error messages here. Is the goal to inspect every key for presence? I thought we just wanted to check the ShopperDefaultLanguage, such as `es`, and determine if they have an `es.json` file at all? Because we don't' have a requirement that every translation key is present in any given language file, as long as they are fine with anything that is missing falling back to english.

File: lib/stencil-start.js [hunk 1]
```diff
@@ -236,6 +269,23 @@ class StencilStart {
...
+        const filesPaths = await this._fsUtils.recursiveReadDir(langsPath);
+        const isDefaultLanguagePresent = filesPaths.some((file) =>
+            file.includes(defaultShopperLanguage),
+        );
+
+        if (!isDefaultLanguagePresent) {
+            this._logger.log(
+                `${
+                    'Warning'.yellow
+                }: lang folder is missing default shopper language: ${defaultShopperLanguage}`,
```
Comment (jmwiese) [inline lib/stencil-start.js:282]: 🍹 I would maybe reword this to be something like "missing language file for default shopper language: ${defaultShopperLanguage}". I think that reads more naturally to me.

File: lib/lang-helper.js [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+                const langFile = await this._fsUtils.parseJsonFile(filePath);
+                const flattenedLang = flattenObject(langFile);
+                const langKeys = Object.keys(flattenedLang);
+                for (const langKey of langKeys) {
+                    if (!alreadyWarnedKey.includes(langKey)) {
+                        if (!flattenedDefaultLang[langKey]) {
+                            this._logger.log(
+                                `${
+                                    'Warning'.yellow
+                                }: file: ${defaultLangFilePath} doesn't have ${langKey}, which is presented in ${filePath}`,
```
Comment (jmwiese) [inline lib/lang-helper.js:42]: 🍹 which is presented in -> which is present in

**Review by jmwiese — APPROVED** [review]

**Comment by bookernath** [comment]
Nice work!

**Review by bookernath — APPROVED** [review]
